typing/printtyp.ml

```diff
@@ -277,10 +277,21 @@ let rec normalize_type_path ?(cache=false) env p =
   with
     Not_found -> (p, Id)
 
+let penality s =
+  if s <> "" && s.[0] = '_' then
+    10
+  else
+    try
+      for i = 0 to String.length s - 2 do
+        if s.[i] = '_' && s.[i + 1] = '_' then
+          raise Not_found
+      done;
+      1
+    with Not_found -> 10
```

Comment:
Also `Exit` is not used by library functions, so it's easier to reason about its potential raise sites -- although in that particular case it's rather trivial as `String.length` does not raise anyway.